### PR TITLE
refactor(client): ux create a project form (#1736)

### PR DESCRIPTION
**Pull request by andre-code**
### Description

This PR contains the update of the New Project creation form to use the new form styles and the progress indicator component.
It also separates the components by file (project/new/components folder), and the non-complex components are rewritten in typescript.

new project form

/deploy renku-core=develop renku=tests-new-project-form-changes #persist

### CODE REVIEW

File: client/src/project/new/ProjectNew.state.js [hunk 1]
```diff
@@ -718,13 +718,13 @@ class NewProjectCoordinator {
 
     // check other errors (requires user intervention). Skip if there is already a warning
     if (!warnings["namespace"] && !input.namespace)
-      errors["namespace"] = "Select namespace.";
+      errors["namespace"] = "Please select namespace";
```
Comment (ciyer) [inline client/src/project/new/ProjectNew.state.js:721]:
```suggestion
      errors["namespace"] = "Please select the namespace for the project";
```

File: client/src/project/new/ProjectNew.state.js [hunk 2]
```diff
@@ -718,13 +718,13 @@ class NewProjectCoordinator {
...
 
     if (!warnings["visibility"] && !input.visibility)
-      errors["visibility"] = "Select visibility.";
+      errors["visibility"] = "Please select visibility";
```
Comment (ciyer) [inline client/src/project/new/ProjectNew.state.js:724]:
```suggestion
      errors["visibility"] = "Please select a visibility level";
```

File: client/src/project/new/ProjectNew.present.js [hunk 3]
```diff
@@ -211,1074 +191,114 @@ function ForkProjectContent(props) {
...
+const NewProjectForm = (
+  { automated, config, handlers, input, meta, namespaces, namespace,
+    user, importingDataset, userRepo, templates } ) => {
+
+  const isFormProcessingOrFinished = (meta) => {
+    // posting
+    if (meta.creation.creating || meta.creation.projectUpdating || meta.creation.kgUpdating)
+      return true;
+    // posted successfully with visibility or KG warning
+    return !!(meta.creation.created && (meta.creation.projectError || meta.creation.kgError));
```
Comment (ciyer) [inline client/src/project/new/ProjectNew.present.js:209]: This is clever, but I would prefer code that is a bit "dumber." :)

```suggestion
    if (meta.creation.created) return true;
    return (meta.creation.projectError || meta.creation.kgError);
```

**Review by ciyer — CHANGES_REQUESTED** [review]
This is a very nice refactoring. Splitting the one monster file into a one component per file is alone very helpful for following and working on this code, but there are many other nice improvements here as well!

I just have a few suggestions for small code readability improvements.

**Comment by andre-code** [comment]
> This is a very nice refactoring. Splitting the one monster file into a one component per file is alone very helpful for following and working on this code, but there are many other nice improvements here as well!
> 
> I just have a few suggestions for small code readability improvements.

Thanks for your suggestions @ciyer 🙌 all were included.
🚀

**Review by ciyer — APPROVED** [review]
🚀 Let us get this merged!

**Pull request merged**